test/oli_web/live/publish_live_test.exs

```diff
@@ -0,0 +1,374 @@
+      })
+
+    insert(:project_resource, %{project_id: project.id, resource_id: container_resource.id})
+
+    publication =
+      insert(:publication, %{
+        project: project,
+        root_resource_id: container_resource.id,
+        published: nil
+      })
+
+    insert(:published_resource, %{
+      publication: publication,
+      resource: container_resource,
+      revision: container_revision,
+      author: author
+    })
+
+    insert(:published_resource, %{
+      publication: publication,
+      resource: page_resource,
+      revision: page_revision,
+      author: author
+    })
+
+    section1 =
+      insert(:section,
+        title: "Example_Section1",
+        base_project: project,
+        type: :enrollable,
+        open_and_free: true,
+        registration_open: true,
+        start_date: yesterday(),
+        end_date: tomorrow()
+      )
+
+    section2 =
+      insert(:section,
+        title: "Example_Section2",
+        base_project: project,
+        type: :enrollable,
+        open_and_free: true,
+        registration_open: true,
+        start_date: yesterday(),
+        end_date: tomorrow()
+      )
+
+    {:ok, _sr} = Sections.create_section_resources(section1, publication)
+    {:ok, _sr} = Sections.create_section_resources(section2, publication)
+
+    [
+      project: project,
+      publication: publication,
+      page_revision: page_revision,
+      section1: section1,
+      section2: section2,
+      author: author,
+      container_revision: container_revision
+    ]
+  end
+
+  defp create_project_with_publication(_conn) do
+    project = insert(:project)
+    container_resource = insert(:resource)
+
+    container_revision =
+      insert(:revision, %{
+        resource: container_resource,
+        resource_type_id: ResourceType.get_id_by_type("container"),
+        children: [],
+        content: %{},
+        deleted: false,
+        slug: "root_container",
+        title: "Root Container"
+      })
+
+    insert(:project_resource, %{project_id: project.id, resource_id: container_resource.id})
+
+    publication =
+      insert(:publication, %{
+        project: project,
+        root_resource_id: container_resource.id,
+        published: nil
+      })
+
+    insert(:published_resource, %{
+      publication: publication,
+      resource: container_resource,
+      revision: container_revision
+    })
+
+    %{project: project}
+  end
+
+  describe "user cannot access when is not logged in" do
+    test "redirects to new session when accessing the publish view", %{conn: conn} do
+      project = insert(:project)
+
+      assert conn
+             |> get(live_view_publish_route(project.slug))
+             |> html_response(302) =~
+               "You are being <a href=\"/authoring/session/new?request_path=%2Fauthoring%2Fproject%2F#{project.slug}%2Fpublish\">redirected</a>"
+    end
+  end
+
+  describe "user cannot access when is logged in as an author but is not a system admin" do
```

Comment:
Not sure if this case makes sense... the author can't access because is not author of the project more than because is not a system admin